test/over_react_test/react_util_test.dart

```diff
@@ -785,22 +821,21 @@ main() {
 
           expect(queryAllByTestId(renderedInstance, 'value'), innerNodes);
         });
+      });
 
-        test('expect when no matching element exists', () {
-          var renderedInstance = render(Dom.div()());
+      group('from multiple layers within nested ShadowRoots when `searchInShadowDom` is `true`', () {
+        test('', () async {
+          var jacket = mount(DeeplyShadowNested()());
 
-          expect(queryAllByTestId(renderedInstance, 'value'), isEmpty);
-        });
-      });
+          // Let the shadow dom mount (the test components kinda slow since it does it after adding it to the dom.)
+          await Future.delayed(const Duration(milliseconds: 500));
 
-      test('custom html attribute key', () {
-        var renderedInstance = render(Wrapper()(
-          (Nested()..addTestId('value'))(),
-          (Nested()..addTestId('value'))(),
-        ));
-        var innerNodes = findDomNode(renderedInstance).querySelectorAll('[data-test-id~="inner"]');
+          var firstShadow = jacket.mountNode.querySelector('[data-test-id~="firstShadow"]');
+          var level2deeplyNested = firstShadow.shadowRoot.children.first.querySelector('[data-test-id~="deeplyNested"]');
+          var level3deeplyNested = firstShadow.shadowRoot.children.first.children.first.shadowRoot.querySelector('[data-test-id~="deeplyNested"]');
 
-        expect(queryAllByTestId(renderedInstance, 'value'), innerNodes);
+          expect(queryAllByTestId(jacket.mountNode, 'deeplyNested', searchInShadowDom: true), [level2deeplyNested, level3deeplyNested]);
```

Comment:
Since the default `depth` is 1, should it have only returned `level2deeplyNested` and not `level3deeplyNested`?

Reply:
hrm...

Reply:
I think your right, i think i messed up and need to put the like depth increment above the foreach